protocol/v2/types/ssvshare.go

```diff
@@ -15,6 +15,11 @@ import (
 	beaconprotocol "github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
 )
 
+const (
+	MaxPossibleShareSize = 1245
+	MaxAllowedShareSize  = MaxPossibleShareSize * 2 // Leaving some room for protocol updates and calculation mistakes.
```

Comment:
i think we can be even more forgiving and allow up to 16kb per share (8 * 1024)

the attack vector they mentioned is if we try to load megabytes of data, so we don't really need to be precise, but rather just avoid huge decoding datas

Reply:
i think we'll likely forget to update these anyway, no matter how we call the consts

alternatively, you can just increase the factor from 2 to 8 for example, just to give more room for mistakes

after all, we don't even think this is a realistic attack vector 😅

Reply:
I just don't understand why we need to tie to an exact number. But I don't mind increasing the factor. With the current approach, on share size increase the test checking the share size would break and we'd immediately see it and adjust the value

Reply:
the bad scenario with small allowable size is that some user creates a share with a long graffiti (or a new field we allow) and we don't test this, but then in production it will not decode, so i prefer to allow big sizes as long as they don't cause us to go OOM

Reply:
I understand your concern but we don't have such fields, the graffiti is 32 bytes. Also, `MaxPossibleShareSize` is calculated by creating a share of max possible size

Reply:
@moshe-blox Increased factor to 8

Reply:
By the way, setting a constant is less readable in my opinion. If someone looks at the 16kb limit, it's not clear if the share fits it and requires calculating the share size